readme.md

```diff
@@ -541,7 +541,330 @@ namespace Azure.Service.Models
+            writer.WriteEndObject();
+        }
+
+        internal static Cat DeserializeCat(JsonElement element)
+        {
+            if (element.ValueKind == JsonValueKind.Null)
+            {
+                return null;
+            }
+            string name = default;
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("name"u8))
+                {
+-                   meow = property.Value.GetString();
++                   DeserializeName(property, ref name);
+                    continue;
+                }
+            }
+            return new Cat(name);
+        }
+    }
+}
+```
+
+</details>
+
+If you want to add a new property to the model and also add the property into the serialization/deserialization methods, you could also use the `CodeGenMemberSerialization` attribute and the `CodeGenMemberSerializationHooks` attribute.
+
+<details>
+
+Define a partial class and add the property you want to add in the partial class, and add `CodeGenMemberSerialization` attribute to it:
+
+``` C#
+public partial class Cat
+{
+    [CodeGenMemberSerialization("color")]
+    public string Color { get; set; } // assuming the original Cat class does not have the Color property
+}
+```
+
+**Generated code after (Generated/Models/Model.Serialization.cs):**
+
+``` diff
+namespace Azure.Service.Models
+{
+    public partial class Cat : IUtf8JsonSerializable
+    {
+        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            writer.WritePropertyName("name"u8);
+            writer.WriteStringValue(Name);
+            writer.WritePropertyName("color"u8);
++           writer.WriteStringValue(Color);
+            writer.WriteEndObject();
+        }
+
+        internal static Cat DeserializeCat(JsonElement element)
+        {
+            if (element.ValueKind == JsonValueKind.Null)
+            {
+                return null;
+            }
+            string name = default;
++           Optional<string> color = default;
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("name"u8))
+                {
+                    meow = property.Value.GetString();
+                    continue;
+                }
++               if (property.NameEquals("color"u8))
++               {
++                   meow = property.Value.GetString();
++                   continue;
++               }
+            }
+            return new Cat(name);
+        }
+    }
+}
+```
+
+You could also add the `CodeGenMemberSerializationHooks` attribute to the property to have your own serialization/deserialization logic of the new property. You might have to do this if the type of your new property is an object type or any type that our generator does not natively support.
+
+**NOTE: Adding property to serialization/deserialization methods currently only works for DPG.**
```

Comment:
Does this PR implement for Mgmt as well?

Reply:
Not yet.

Reply:
OK. I thought the change in `test/TestProjects/MgmtCustomizations/Customization/` is showing it works for mgmt.

Reply:
Oh - I forgot to remove it. I will add a comment to that property emphasizing that this does not work now.

Reply:
Actually i already removed the `OldProperty` which is expected to be added into the serialization and deserialization.

Reply:
Just to make sure I understand this works for all targets right?

Reply:
> Just to make sure I understand this works for all targets right?

Adding an extra property to serialization/deserialization using the attribute only works for DPG.